clarity/src/vm/analysis/errors.rs

```diff
@@ -493,3 +495,52 @@ impl DiagnosableError for CheckErrors {
         }
     }
 }
+
+impl From<CodecError> for CheckErrors {
```

Comment:
Instead of doing this, why not just create a single `CheckErrors::Codec(ce: CodecError)` variant?  Why attempt to add more variants to `CheckErrors` that only serve to represent (some) variants of `CodecError`?